pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -189,12 +226,7 @@ func createNamedClusterInfo(context *clusterd.Context, namespace string) (*cephc
 	}
 
 	// generate the admin secret if one was not provided at the command line
-	args := []string{
-		"--cap", "mon", "'allow *'",
-		"--cap", "osd", "'allow *'",
-		"--cap", "mgr", "'allow *'",
-		"--cap", "mds", "'allow'"}
-	adminSecret, err := genSecret(context.Executor, dir, cephclient.AdminUsername, args)
+	adminSecret, err := genSecret(context.Executor, dir, cephclient.AdminUsername, adminCapArgs)
```

Comment:
With the new approach we need to first generate the `client.admin` keyring
```suggestion
	adminSecret, err := genSecret(context.Executor, dir, cephclient.NonOperatorAdminUsername, adminCapArgs)
```